src/CoreBundle/private/library/classes/dbobjects/TCMSRecord.class.php

```diff
@@ -884,6 +884,11 @@ public function &GetTableConf()
             } else {
                 $this->_oTableConf->LoadFromFieldWithCaching('name', $this->table);
             }
+
+            // TODO alternative 2 - but probably too late ($this->_oTableConf is cached)
+            if (null !== $this->iLanguageId) {
```

Comment:
We can set the language in line 873 (`GetNewInstance(null, $this->iLanguageId)`) - there the null case is also already handled. I think this could replace all three changes in this PR.

Reply:
done